cmd/mt-store-cat/chunk.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/grafana/metrictank/store/cassandra"
+	log "github.com/sirupsen/logrus"
+)
+
+// printChunkSummary prints a summary of chunks in the store matching the given conditions, grouped in buckets of groupTTL size by their TTL
+func printChunkSummary(ctx context.Context, store *cassandra.CassandraStore, tables []cassandra.Table, metrics []Metric, groupTTL string) error {
+	now := uint32(time.Now().Unix())
+	end_month := now - (now % cassandra.Month_sec)
+
+	for _, tbl := range tables {
+		// per store.FindExistingTables(), actual TTL may be up to 2x what's in tablename.
+		// we query up to 4x so that we also include data that should have been dropped already but still sticks around for whatever reason.
+		start := now - uint32(4*tbl.TTL*3600)
+		start_month := start - (start % cassandra.Month_sec)
+		fmt.Println("## Table", tbl.Name)
+		if len(metrics) == 0 {
+			query := fmt.Sprintf("select key, ttl(data) from %s", tbl.Name)
+			iter := store.Session.Query(query).Iter()
+			showKeyTTL(iter, groupTTL)
+		} else {
+			for _, metric := range metrics {
+				for month := start_month; month <= end_month; month += cassandra.Month_sec {
+					row_key := fmt.Sprintf("%s_%d", metric.AMKey.String(), month/cassandra.Month_sec)
+					query := fmt.Sprintf("select key, ttl(data) from %s where key=?", tbl.Name)
+					iter := store.Session.Query(query, row_key).Iter()
+					showKeyTTL(iter, groupTTL)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func printChunkCsv(ctx context.Context, store *cassandra.CassandraStore, table cassandra.Table, metrics []Metric, start, end uint32) {
+
+	// see CassandraStore.SearchTable for more information
+	start_month := start - (start % cassandra.Month_sec)       // starting row has to be at, or before, requested start
```

Comment:
Possible I just don't get it why this is necessary... but:
`start_month` is only used 2 lines down, where it will be divided by `cassandra.Month_sec`. Since integer divisions would drop the `start % cassandra.Month_sec` anyway, how is this code different from:

`startMonthNum := start / cassandra.Month_sec`

Reply:
that sounds good.
i suppose similarly we can also do:
```
endMonthNum := (end - 1) / cassandra.Month_sec
```
right?
note: this code comes from the cassandra code, and i think we do it elsewhere too. so perhaps better to do a separate cleanup PR for just that.

Reply:
yeah i think you're right. i wasn't completely sure by using logic, so i just tested it: https://play.golang.org/p/DOl02-EibSU

Reply:
-> #1148